Mathlib/Topology/UniformSpace/Basic.lean

```diff
@@ -141,6 +141,32 @@ theorem idRel_subset {s : Set (α × α)} : idRel ⊆ s ↔ ∀ a, (a, a) ∈ s
   simp [subset_def]
 #align id_rel_subset idRel_subset
 
+theorem prod_subset_idRel_Eq_singleton_left {X : Type _} {S T : Set X} (hS : S.Nonempty)
```

Comment:
Do you really need the left and right versions after proving the stronger version?

Reply:
Well, I use the left to prove the right and both to prove the full one. For my application I only need the left (for example) but I find it more complete to have the three.